xla/service/gpu/gpu_float_support.cc

```diff
@@ -21,7 +21,6 @@ namespace gpu {
 bool GpuFloatSupport::IsSupported(const HloInstruction& hlo) const {
   switch (hlo.opcode()) {
     // Collective ops.
-    case HloOpcode::kAllGather:
```

Comment:
It seems for all-gather, all-to-all, and collective-permute, all types are always supported, so should we change it that way? And only for all-reduce/reduce-scatter and variants, check the LowPrecisionType()?

Reply:
Since this PR only targets at AllGather. Do you mean return true for AllGather regardless of the low precision type?

Reply:
Yeah, except for all-reduce and reduce-scatter (and their async forms), return true for all other collectives since they just move data and can be casted from their original type to same sized int, regardless of low precision type.